lucet-concurrency-tests/tests/killswitch.rs

```diff
@@ -0,0 +1,114 @@
+use lucet_runtime::{Error, KillSuccess, Limits, Region, TerminationDetails};
+use std::sync::Arc;
+use std::thread;
+
+use lucet_module::FunctionPointer;
+use lucet_runtime::MmapRegion;
+use lucet_runtime_internals::lock_testpoints::Syncpoint;
+use lucet_runtime_internals::module::Module;
+use lucet_runtime_internals::module::{MockExportBuilder, MockModuleBuilder};
+use lucet_runtime_internals::vmctx::lucet_vmctx;
+
+static mut ENTERING_GUEST: Option<Syncpoint> = None;
```

Comment:
I tried making this a `thread_local` only to eventually realize that actually doesn't cut it - `inst.run` happens in a different thread than the initial setup, so the test thread and instance thread have different `Syncpoint` :(